app/controllers/finance/financial_links_controller.rb

```diff
@@ -64,6 +64,23 @@ def remove_bank_transaction
     redirect_to finance_link_url(@financial_link), notice: t('.notice')
   end
 
+  def new_financial_transaction
+    @financial_transaction = FinancialTransaction.new(financial_link: @financial_link)
+    @financial_transaction.amount = @financial_link.amount
+    @financial_transaction.ordergroup_id = FinancialTransaction.joins(<<-SQL).order(created_on: :desc).pluck(:ordergroup_id).first
+      JOIN bank_transactions a ON financial_transactions.financial_link_id = a.financial_link_id
+      JOIN bank_transactions b ON a.iban = b.iban AND b.financial_link_id = #{@financial_link.id}
+    SQL
```

Comment:
This would be a candidate for moving to a model or another business-logic-object.
I'd put determining the ordergroup in a separate method somewhere, so that it can be more easily overridden (as it looks like a bit of a heuristic).

Reply:
done